### PR TITLE
feat(tae): collect data and columns log tail

### PR DESCRIPTION
**Pull request by aptend**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #4678 

## What this PR does / why we need it:
- Collect data change log tail
- Collect columns change log tail

### CODE REVIEW

File: pkg/vm/engine/tae/db/logtail.go [hunk 1]
```diff
@@ -603,40 +628,71 @@ func (b *TableLogtailRespBuilder) visitBlkMeta(e *catalog.BlockEntry) {
...
 
-func (b *TableLogtailRespBuilder) visitBlkData(e *catalog.BlockEntry) {
-	_ = e.GetBlockData()
+func (b *TableLogtailRespBuilder) visitBlkData(e *catalog.BlockEntry) (err error) {
+	block := e.GetBlockData()
+	insBatch, err := block.CollectAppendInRange(b.start, b.end)
+	if err != nil {
+		return
+	}
+	if insBatch.Length() > 0 {
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/db/logtail.go:641]: insBatch may be nil here

File: pkg/vm/engine/tae/db/logtail.go [hunk 2]
```diff
@@ -603,40 +628,71 @@ func (b *TableLogtailRespBuilder) visitBlkMeta(e *catalog.BlockEntry) {
...
+		extendBatch(b.dataInsBatch, insBatch)
+	}
+	delBatch, err := block.CollectDeleteInRange(b.start, b.end)
+	if err != nil {
+		return
+	}
+	if delBatch.Length() > 0 {
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/db/logtail.go:648]: Maybe nil

**Review by XuPeng-SH — CHANGES_REQUESTED** [review]

File: pkg/vm/engine/tae/db/logtail.go [hunk 3]
```diff
@@ -555,6 +563,14 @@ func containersBatchToProtoBatch(batch *containers.Batch) (*api.Batch, error) {
 	return protoBatch, nil
 }
 
+// extend vector with same name, consume src batch
+func extendBatch(dst, src *containers.Batch) {
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/db/logtail.go:567]: why not add a Extend api for batch?

File: pkg/vm/engine/tae/db/logtail.go [hunk 4]
```diff
@@ -575,15 +591,24 @@ func NewTableLogtailRespBuilder(ckp string, start, end types.TS, tbl *catalog.Ta
...
 	b.dataInsBatch = makeRespBatchFromSchema(b.dataInsSchema)
 
 	b.did = tbl.GetDB().GetID()
 	b.tid = tbl.ID
 	b.dname = tbl.GetDB().GetName()
 	b.tname = b.dataInsSchema.Name
 
-	// TODO: dataDelBatch
+	// dataDelBatch, return rowid anyway, if has pk, carry pk too
+	dataDelSchema := catalog.NewEmptySchema("dataDel")
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/db/logtail.go:605]: why pk is needed？

File: pkg/vm/engine/tae/db/logtail.go (hunk 1 above)
Comment (aptend) [reply pkg/vm/engine/tae/db/logtail.go]: checked

File: pkg/vm/engine/tae/db/logtail.go (hunk 2 above)
Comment (aptend) [reply pkg/vm/engine/tae/db/logtail.go]: checked

File: pkg/vm/engine/tae/db/logtail.go (hunk 3 above)
Comment (aptend) [reply pkg/vm/engine/tae/db/logtail.go]: done

File: pkg/vm/engine/tae/db/logtail.go (hunk 4 above)
Comment (aptend) [reply pkg/vm/engine/tae/db/logtail.go:597]: pk is removed from CollectDelete

**Review by XuPeng-SH — APPROVED** [review]
lgtm

**Review by XuPeng-SH — APPROVED** [review]
lgtm

**Pull request merged**